### PR TITLE
Removed fuel from encoding of Dafny opaque functions

### PR DESCRIPTION
**Pull request by EkanshdeepGupta**
At present, for encoding opaque functions, Dafny sets it with 0 fuel, and inserts a boolean flag which guards whether the body of the opaque function is available or not. These two mechanisms essentially perform the same function and thus are redundant.

This PR is an attempt to make the dafny encoding of opaque functions leaner by getting rid of the fuel in Dafn'y encoding of opaque functions. 

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs
Comment (MikaelMayer) [inline Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs:74]:
```suggestion
```

File: Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs
Comment (MikaelMayer) [inline Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs:105]: Please remove this comment

File: Source/DafnyCore/Verifier/Translator.ExpressionTranslator.cs
Comment (MikaelMayer) [inline Source/DafnyCore/Verifier/Translator.ExpressionTranslator.cs:1922]: Could you change this attribute name to something more explicit, such as "revealedFunction" ?

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
Nice. I'm curious about the variability of this change.

File: Source/DafnyCore/Verifier/Translator.ExpressionTranslator.cs
Comment (EkanshdeepGupta) [reply Source/DafnyCore/Verifier/Translator.ExpressionTranslator.cs:1922]: For sure.

File: Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs
Comment (EkanshdeepGupta) [reply Source/DafnyCore/Rewriters/OpaqueMemberRewriter.cs]: Done

**Review by MikaelMayer — APPROVED** [review]

**Comment by keyboardDrummer** [comment]
Should we get a second approver for these changes?

**Comment by atomb** [comment]
I think @MikaelMayer has noticed some performance impact that needs investigating, so let's not merge this quite yet. I'm happy to be the second reviewer once that's done.

**Comment by seanmcl** [comment]
Thanks @EkanshdeepGupta.  Fantastic work.

**Comment by EkanshdeepGupta** [comment]
Oh this made its way into master? I am curious, there were all kinds of failures in the integration tests when I tried it. What changed?

**Comment by MikaelMayer** [comment]
I fixed all issues :-)

**Pull request merged**
